client/src/components/pub-search.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { Component } from 'react';
+import { ApiService } from '../services';
+import { navigate, Link } from '@reach/router';
+
+class PubSearch extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      twitterHandle: '',
+      searchTitle: '',
+      searchTwitterHandle: '',
+      data: [],
+    };
+  }
+
+  handleInputChange = (event) => {
+    const { name, value } = event.target;
+    this.setState({
+      [name]: value,
+    });
+  };
+
+  handleChange = (event) => {
+    let target = event.target;
+    let name = target.name;
+    this.setState({
+      [name]: event.target.value,
+    });
+  };
+
+  handleCancel = (event) => {
+    event.preventDefault();
+    navigate('/pub-display');
+  };
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+    let query = '';
+    let pubName = this.state.searchTitle;
+    let pubTwitter = this.state.searchTwitterHandle;
+    if (pubName.length > 0) {
+      query = 'pubs/searchPubTitle/' + pubName;
+    } else {
+      query = 'pubs/searchPubTwitter/' + pubTwitter;
+    }
+    if (this.state.searchTitle.length === 0 && this.state.searchTwitterHandle.length === 0) {
+      alert('Please enter either a publication title or a Twitter handle to run the search');
+      navigate('/pub-search');
+    } else {
+      ApiService.get(query)
+        .then((response) => {
+          if (response.data === null) {
+            alert('No publication with that name found. Please enter it into the database.');
+            navigate('/pub-entry');
+          } else {
+            this.setState({
+              data: response.data,
+            });
+          }
+        })
```

Comment:
This makes sense in theory. Do I understand correctly that because `pub-search.jsx` is a child of `pub-display.jsx` (where the original API call takes place), I can access the `pub-display` data using `props`? I'll have to figure that out...

Reply:
One solution to persisting state across pages is to elevate the data to a higher (global) component. The easiest spot is `App.jsx`. When the API call is executed in `pub-display`, then call a props function `updatePublications` (which is passed down from `App.jsx`) with the data received from the API call.

This does not have to be done here and now in this PR. Executing search via API calls is a great starting point! We can enhance (or not) in the future. 😄